processor/src/main/java/org/mapstruct/ap/internal/model/beanmapping/SourceReference.java

```diff
@@ -174,6 +184,23 @@ private SourceReference buildFromSingleSourceParameters(String[] segments, Param
             return new SourceReference( parameter, entries, foundEntryMatch );
         }
 
+        private SourceReference forMapSource(String[] segments, Parameter parameter) {
+            Type valueType = null;
+            final List<Type> typeParameters = parameter.getType().getTypeParameters();
+            if (typeParameters.size() == 2) {
+                valueType = typeParameters.get(1);
+            } else {
+                //TODO: this should be a type mirror of Object by default
```

Comment:
@sjaakd Do you have any advice on how to obtain a TypeMirror for Java's Object for providing a default if the Map reference is untyped?

Reply:
there is a method on `TypeFactory`.. `Type getType(Class<?> type)`.. You mean something along the lines like `typeFactory.getType( Object.class )`